packages-exp/firebase-exp/rollup.config.js

```diff
@@ -121,6 +121,43 @@ const appBuilds = [
   }
 ];
 
+const messagingBuilds = [
+  /**
+   * Messaging Browser Builds
+   */
+  {
+    input: 'messaging/index.ts',
+    output: [
+      {
+        file: resolve('messaging', appPkg.main),
+        format: 'cjs',
+        sourcemap: true
+      },
+      {
+        file: resolve('messaging', appPkg.module),
+        format: 'es',
+        sourcemap: true
+      }
+    ],
+    plugins: [...plugins, typescriptPlugin],
+    external
+  },
+
+  /**
+   * Messaging UMD Builds. Required for FM SDK to work but sw does yet support modules.
+   */
+  {
+    input: 'messaging/index.cdn.ts',
+    output: {
+      file: 'firebase-messaging.js',
+      sourcemap: true,
+      format: 'umd',
+      name: `${GLOBAL_NAME}.app`
+    },
+    plugins: [...plugins, typescriptPluginUMD, uglify()]
```

Comment:
If you decide to keep messaging as a separately defined build set, the only other difference I noticed is the external line isn't here, is there a reason?